connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoIngestionUtils.scala

```diff
@@ -38,12 +37,12 @@ object KustoIngestionUtils {
   }
 
   private[kusto] def setCsvMapping(sourceSchema: StructType,
-                                   targetSchema: Array[JSONObject],
+                                   targetSchema: Array[JsonNode],
                                    ingestionProperties: SparkIngestionProperties,tableCreationMode: SinkTableCreationMode): Unit = {
     require(ingestionProperties.csvMappingNameReference == null || ingestionProperties.csvMappingNameReference.isEmpty,
       "Sink options SparkIngestionProperties.csvMappingNameReference and adjustSchema.GenerateDynamicCsvMapping are not compatible. Use only one.")
 
-    val targetSchemaColumns = targetSchema.map(c => (c.getString("Name"),c.getString("CslType"))).toMap
+    val targetSchemaColumns = targetSchema.map(c => (c.get("Name").asText(),c.get("CslType").asText())).toMap
```

Comment:
Same as comment above, the constant strings

Reply:
Done